cpp/src/arrow/test-util.h

```diff
@@ -81,25 +83,20 @@ namespace test {
 
 template <typename T>
 void randint(int64_t N, T lower, T upper, std::vector<T>* out) {
-  Random rng(random_seed());
-  uint64_t draw;
-  uint64_t span = upper - lower;
-  T val;
-  for (int64_t i = 0; i < N; ++i) {
-    draw = rng.Uniform64(span);
-    val = static_cast<T>(draw + lower);
-    out->push_back(val);
-  }
+  const int random_seed = 0;
+  std::mt19937 gen(random_seed);
```

Comment:
Random or static seed?

Reply:
This was originally calling the `random_seed()` function which just returned zero. I can add a seed parameter.